api/proto/config/appconfig.proto

```diff
@@ -118,6 +118,14 @@ message AppInstanceConfig {
   // be started independent of the global or local profile specified for the
   // device.
   repeated string profile_list = 18;
+
+  // The static IP address assigned on the NetworkAdpater of the App which serves
+  // as the local data store server. If this 'localDataStoreIPAddr' is valid,
+  // it enables the device datastore config where the FQDN IP address matches this
+  // address and to be routed locally inside the device. If a Port map is enabled
+  // for external access, that would be a normal inbound access operation for
+  // this App to serve the data store remotely also.
+  string localDataStoreIPAddr = 19;
```

Comment:
In principle, can't this just be a boolean "app is providing local services to EVE" and the IP address is in the static IP for the app instance network adapter? In theory the app could even be assigned a dynamic IP but that is not very useful in practice since the app needs to be deployed before the datastore etc can be created in the controller. But from a routing/source address/PBR perspective, as long as zedrouter/downloader etc knows the app should be accessible they can compare e.g., the datastore config to see if it matches any of the apps with the boolean set.

Reply:
I'm a very big +1 on it being a boolean, even more so -- I think it highlight a need for a bit more of a design proposal added to the wiki that we can all iterate on. @naiming-zededa -- do you think you can do that first before we decide whether to merge this one or not?